demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -50,523 +135,127 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-                    cv::imshow(imshowWindowTitle, requestResult.frame);
-
-                    const int key = cv::waitKey(1);
-
-                    if (27 == key || 'q' == key || 'Q' == key) {  // Esc
-                        break;
-                    } else if (9 == key) {  // Tab
-                        if (prevModeActiveRequestCount != 0) continue;
-
-                        ExecutionMode prevMode = currentMode;
-                        switchMode(currentMode);
-
-                        {
-                            std::lock_guard<std::mutex> lock(mutex);
-
-                            prevModeActiveRequestCount = prevMode == ExecutionMode::USER_SPECIFIED
-                                                         ? userSpecifiedInferRequests.size() - emptyRequests.size()
-                                                         : 1 - emptyRequests.size();
-                        }
-
-                        putHighlightedText(requestResult.frame, "Switching modes, please wait...",
-                                           cv::Point2f(10, requestResult.frame.rows - 60),
-                                           cv::FONT_HERSHEY_TRIPLEX, 0.75, cv::Scalar(10, 200, 10), 2);
-                        cv::imshow(imshowWindowTitle, requestResult.frame);
-                        cv::waitKey(1);
-
-                        if (prevMode == ExecutionMode::USER_SPECIFIED) {
-                            for (const auto& request : userSpecifiedInferRequests) {
-                                request->Wait(IInferRequest::WaitMode::RESULT_READY);
-                            }
-                        } else minLatencyInferRequest->Wait(IInferRequest::WaitMode::RESULT_READY);
-
-                        {
-                            std::lock_guard<std::mutex> lock(mutex);
-
-                            if (currentMode == ExecutionMode::USER_SPECIFIED) {
-                                emptyRequests.assign(userSpecifiedInferRequests.begin(),
-                                                     userSpecifiedInferRequests.end());
-                            } else emptyRequests = {minLatencyInferRequest};
-                        }
-
-                        if (modeMetrics.find(currentMode) == modeMetrics.end()) {
-                            modeMetrics.insert({currentMode, PerformanceMetrics()});
-                        } else modeMetrics[currentMode] = PerformanceMetrics();
-                    } else {
-                        presenter.handleKey(key);
-                    }
+            //--- Checking for results and rendering data if it's ready
+            //--- If you need just plain data without rendering - check for getProcessedResult() function
+            std::unique_ptr<ResultBase> result;
+            while (result = pipeline.getResult()) {
+                cv::Mat outFrame = model->renderData(result.get());
+                //--- Showing results and device information
+                if (!outFrame.empty() && !FLAGS_no_show) {
```

Comment:
You should call `presenter.drawGraphs()` and `paintInfo()` even if FLAGS_no_show is false. And you shouldn't call `cv::waitKey(1)` if `FLAGS_no_show` is true.

Reply:
cv:waitKey - fixed, thanks for pointing. But why should we use drawGraphs and paintInfo in case of FLAGS_no_show ? that's meaningless and will confuse customers reading this code.

Reply:
We want to test them as well.

Reply:
changed

Reply:
You haven't moved `cv::waitKey(1)`.
BTW you don't need to check `outFrame` emptiness. You already did that when it was named `curr_frame`.

Reply:
I've changed logic for cv::waitKey  (moved it inside the cycle to actually show every frame in window). as for checking outFrame - I still have to check it, as empty outFrame is a sign of incorrect processing or no frame available. In theory it might be empty even if we have valid input frame.

Reply:
> In theory it might be empty even if we have valid input frame.

How is that possible?

Reply:
changed